twitter-clone/src/pages/LoginPage.tsx

```diff
@@ -0,0 +1,123 @@
+import { useContext, useState, BaseSyntheticEvent } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { AuthContext } from "../AuthContext";
+import AuthForm from "../components/AuthForm";
+import { API_URL } from "../constants";
+
+interface FormData {
+  email: string;
+  password: string;
+}
+
+const LoginPage = () => {
+  const { logIn } = useContext(AuthContext);
+  const navigate = useNavigate();
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+
+  const handleLogin = async (
```

Comment:
Better to store such API-call function in separate helpers file

Reply:
Moved http calls to separate file.